compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3512,6 +3512,145 @@ TR::Register *OMR::Power::TreeEvaluator::vnegDoubleHelper(TR::Node *node, TR::Co
    return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvnegdp);
    }
 
+TR::Register *OMR::Power::TreeEvaluator::vabsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   switch(node->getDataType().getVectorElementType())
+     {
+     case TR::Int8:
+       return TR::TreeEvaluator::vabsInt8Helper(node, cg);
+     case TR::Int16:
+       return TR::TreeEvaluator::vabsInt16Helper(node, cg);
+     case TR::Int32:
+       return TR::TreeEvaluator::vabsInt32Helper(node, cg);
+     case TR::Int64:
+       return TR::TreeEvaluator::vabsInt64Helper(node, cg);
+     case TR::Float:
+       return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvabssp);
+     case TR::Double:
+       return TR::TreeEvaluator::inlineVectorUnaryOp(node, cg, TR::InstOpCode::xvabsdp);
+     default:
+       TR_ASSERT(false, "unrecognized vector type %s\n", node->getDataType().toString()); return NULL;
+     }
+   }
+
+TR::Register *OMR::Power::TreeEvaluator::vabsInt8Helper(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
This function and its "siblings" are nearly identical except for different instruction opcode choices in a couple of places.  In the interests of avoiding duplication and reducing .so size, can they be collapsed into a single implementation where the opcode choices are made from a table based on element type?